src/Utils/Utils.php

```diff
@@ -435,6 +441,6 @@ public static function flat(array $array): array
             $flattened[] = $a;
         });
 
-        return $flattened;
+        return array_unique($flattened, SORT_REGULAR);
```

Comment:
A unique check violates the description of the docblock. It shouldn't be the responsibility of this method to sort out any non-unique values. The user of this method should be responsible for this.